### PR TITLE
src/CoreAudio: update CoreAudio API, remove deprecated API calls.

### PR DESCRIPTION
**Pull request by TerryGeng**
Separated from #4250 :D

### CODE REVIEW

File: src/mumble/CoreAudio.cpp [hunk 1]
```diff
@@ -37,7 +37,7 @@ class CoreAudioInit : public DeferInit {
 		CoreAudioInputRegistrar *cairReg;
 		CoreAudioOutputRegistrar *caorReg;
 	public:
-		CoreAudioInit() : cairReg(NULL), caorReg(NULL) {}
+		CoreAudioInit() : cairReg(nullptr), caorReg(nullptr) {}
```
Comment (Krzmbrzl) [inline src/mumble/CoreAudio.cpp:40]: While we're at it, let's remove this constructor's implementation and set the defaults in the header file.

File: src/mumble/CoreAudio.cpp [hunk 2]
```diff
@@ -84,33 +84,46 @@ const QHash<QString, QString> CoreAudioSystem::getDevices(bool input) {
...
-	err = AudioHardwareGetProperty(kAudioHardwarePropertyDevices, &len, devs);
+	err = AudioObjectGetPropertyData(kAudioObjectSystemObject, &propertyAddress, 0, nullptr, &len, devs);
 	if (err != noErr)
 		return qhReturn;
 
+	if (input) {
+		propertyAddress.mScope = kAudioDevicePropertyScopeInput;
+	} else {
+		propertyAddress.mScope = kAudioDevicePropertyScopeOutput;
+	}
```
Comment (Krzmbrzl) [inline src/mumble/CoreAudio.cpp:108]: Normally I'm not a big fan of 1-liners but in the case of a conditional assignment, I think they are actually more readable:
```suggestion
	propertyAddress.mScope = input ? kAudioDevicePropertyScopeInput : kAudioDevicePropertyScopeOutput;
```

File: src/mumble/CoreAudio.cpp [hunk 3]
```diff
@@ -491,8 +524,8 @@ CoreAudioOutput::CoreAudioOutput() {
 	desc.componentFlags = 0;
 	desc.componentFlagsMask = 0;
 
-	comp = FindNextComponent(NULL, &desc);
-	if (comp == NULL) {
+	comp = FindNextComponent(nullptr, &desc);
+	if (comp == nullptr) {
```
Comment (Krzmbrzl) [inline src/mumble/CoreAudio.cpp:528]: I guess the "idiomatic" way of doing this check would be
```suggestion
	if (!comp) {
```

Ref.: https://stackoverflow.com/questions/3825668/checking-for-null-pointer-in-c-c

File: src/mumble/CoreAudio.h [hunk 4]
```diff
@@ -10,6 +10,9 @@
 #include "AudioOutput.h"
 
 #include <AudioToolbox/AudioToolbox.h>
+#include <Carbon/Carbon.h>
+
+#include "Global.h"
```
Comment (davidebeatrici) [inline src/mumble/CoreAudio.h:15]: Why did you move the includes to the header?

Comment (TerryGeng) [reply src/mumble/CoreAudio.h:15]: Because I move the declaration of two classes to the header file and DeferInit is defined in Global.h. Actually Carbon.h is not necessary to import from the header file.

Comment (davidebeatrici) [reply src/mumble/CoreAudio.h:15]: All clear, thank you very much for your contribution!
